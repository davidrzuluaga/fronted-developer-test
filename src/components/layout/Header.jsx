import React from 'react';
import PropTypes from 'prop-types';

import Navigation from './Navigation';
import {Link } from "react-router-dom";

const Header = ({ headerInfo }) => {

    const { links } = headerInfo.menu;
    return (
        <header>
            <nav className="navbar navbar-expand-lg navbar-light bg-light">
                <Link to="/">
                    {(headerInfo.brand.logo) && <img src={headerInfo.brand.logo} alt="Js Love" width="50px"/>}
                </Link>
                <button
                    className="navbar-toggler"
                    type="button"
                    data-toggle="collapse"
                    data-target="#nav-project"
                    aria-controls="nav-project"
                    aria-expanded="false"
                    aria-label="Toggle navigation"
                >
                    <span className="navbar-toggler-icon"/>
                </button>
                <Navigation links={links}/>
            </nav>
        </header>
    )
};

Header.propTypes = {
    headerInfo: PropTypes.object
};

export default Header;
