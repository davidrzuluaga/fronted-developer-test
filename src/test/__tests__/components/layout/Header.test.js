import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import projectInfo from "../../../__mocks__/projectInfoMock";
import Header from "../../../../components/layout/Header";

describe('Tests <Header />', () => {

    const { headerInfo } = projectInfo;

    test('Render component', () => {
        const header = shallow(<Header headerInfo={headerInfo}/>);
        expect(header.length).toEqual(1);
    });

    test('Check UI not change', () => {
        const header = create(<Header headerInfo={headerInfo}/>);
        expect(header.toJSON()).toMatchSnapshot();
    });

    test('Check props', () => {
        const wrapper = mount(<Header headerInfo={headerInfo}/>);
        expect(wrapper.props().headerInfo).toEqual(headerInfo)
    });

});
