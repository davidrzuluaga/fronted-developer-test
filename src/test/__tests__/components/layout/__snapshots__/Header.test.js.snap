// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests <Header /> Check UI not change 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Js Love"
        src="http:sdakdshkjhasjkdhas"
        width="50px"
      />
    </a>
    <button
      aria-controls="nav-project"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#nav-project"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="nav-project"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
